src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -889,6 +887,11 @@ private bool InferNamedPipesInformation()
                         return false;
                     }
 
+                    if(tokensByBackSlash[4].StartsWith(NamedPipeInstanceNameHeader))
+                    {
+                        InstanceName = tokensByBackSlash[4].Replace(NamedPipeInstanceNameHeader, "");
```

Comment:
Instead of a Repalce, can you use `tokensByBackSlash[4].SubString(NamedPipeInstanceNameHeader.Length);` ?